pydantic/_internal/_model_construction.py

```diff
@@ -569,10 +583,6 @@ def complete_model_class(
         PydanticUndefinedAnnotation: If `PydanticUndefinedAnnotation` occurs in`__get_pydantic_core_schema__`
             and `raise_errors=True`.
     """
-    if config_wrapper.defer_build:
-        set_model_mocks(cls)
-        return False
```

Comment:
Moved in `ModelMetaclass.__new__`, as it makes more sense to be there (we also don't need to build a different config wrapper with a hardcoded `defer_build=True` value in `model_rebuild()`).